Gordon360/Services/NewsService.cs

```diff
@@ -240,21 +238,43 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
 
             if (newData.Image != null)
             {
-                string fileName = newsItem.SNID + ".jpg";
-                string imagePath = NewsUploadsPath + fileName;
-                ImageUtils.UploadImage(imagePath, newData.Image);
-                newsItem.Image = imagePath;
+                // ImageUtils.GetImageFormat checks whether the image type is valid (jpg/jpeg/png)
+                var (extension, format, data) = ImageUtils.GetImageFormat(newData.Image);
+
+                // If there's already an image attached to the news post, simply overwrite the old image
+                // data when uploading a new one
+                if (newsItem.Image != null)
+                {
+                    var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", Path.GetFileName(newsItem.Image));
+                    ImageUtils.UploadImage(imagePath, newData.Image);
+                }
+
+                // If there isn't an image attached to the news post, do the same procedure to save
+                // image in Submit method
+                else
+                {
+                    // Use a unique alphanumeric GUID string as the file name
+                    var filename = $"{Guid.NewGuid().ToString("N")}.{extension}";
+                    var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", filename);
+
+                    var serverAddress = _serverUtils.GetAddress();
+                    if (serverAddress is not string) throw new Exception("Could not upload Student News Image: Server Address is null");
+
+                    var url = $"{serverAddress}/browseable/uploads/{filename}";
+
+                    ImageUtils.UploadImage(imagePath, newData.Image);
+
+                    newsItem.Image = url;
+                }
             }
 
             //If the image property is null, it means that either the user
             //chose to remove the previous image or that there was no previous
             //image (DeleteImage is designed to handle this).
-            else
+            else if (newsItem.Image != null)
             {
-                string fileName = newsItem.SNID + ".jpg";
-                string imagePath = NewsUploadsPath + fileName;
-                ImageUtils.DeleteImage(imagePath);
-                newsItem.Image = newData.Image;//null
+                ImageUtils.DeleteImage(newsItem.Image);
```

Comment:
Same thing here as above. I'm not sure that ImageUtils.DeleteImage can take in a URL.

Reply:
See comment on above.